lib/Sema/ConstraintSystem.h

```diff
@@ -1452,9 +1452,10 @@ class ConstraintSystem {
   /// Get the type for an expression.
   Type getType(const Expr *E) const {
     assert(hasType(E) && "Expected type to have been set!");
-    assert((!E->getType() ||
-            E->getType()->isEqual(ExprTypes.find(E)->second)) &&
-           "Mismatched types!");
+    // FIXME: lvalue differences
```

Comment:
Can you make this assert more precise instead of commenting it out? Or is this all going away soon anyway?